doc/api/packages.md

```diff
@@ -0,0 +1,660 @@
+# Modules: Packages
+
+<!-- type=misc -->
+
+## Introduction
+
+A package is a folder described by a `package.json` file.
+
+A folder containing a `package.json` file, and all subfolders below that folder
+until the next folder containing another `package.json` file, are a _package
+scope_.
```

Comment:
The package scope concept isn’t mentioned elsewhere on this page. Should we move the “Package scope and file extensions” section, and possibly other parts of the ESM enabling section, here? All the content about package scope and `"type"` applies equally to CommonJS as ESM.

Reply:
There's a section for terms, right?

```html
<dt>Package scope</dt>
<dd>

folder containing a `package.json` file, and all subfolders below that folder
until the next folder containing another `package.json`

</dd>
```

Reply:
Looks like we still have everything over at the following URL.

https://github.com/nodejs/modules/blob/master/doc/terminology.md

Should we add this definition to there?

Reply:
> Should we add this definition to there?

That document was written to help guide the modules group in coming up with the new implementation. It isn't part of the Node docs.

The docs need to explain package scope in order to explain `"type"`. So it needs to be in the docs. I'm thinking that the whole Enabling section needs to move to the Packages page, and the intro sentences rewritten from the perspective of “how to enable ESM” to instead be “how a package tells Node whether it’s CommonJS or ESM.” And then the ESM page still needs an Enabling section, something like:

> ## Enabling
>
> Node.js treats JavaScript code as CommonJS modules by default. Authors can tell Node.js to treat JavaScript code as ECMAScript modules via the `.mjs` file extension, the `package.json` `"type"` field, or the `--input-type=module` flag. See [Determining Module System]().

Where the link at the end links to what was the Enabling section, now living on the Packages page. And the above brief paragraph is all we need as a new Enabling section in the ESM page.

Reply:
Done in 7043be8a2172b4d16ee2b348d51c7197087f1dec, PTAL